rocAL/include/decoders/image/rocjpeg_decoder.h

```diff
@@ -0,0 +1,139 @@
+/*
+Copyright (c) 2025 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#pragma once
+#include "decoders/image/decoder.h"
+#include "rocjpeg.h"
+#include "decoders/image/rocjpeg_decoder.h"
+
+struct ScalingFactor {
+  unsigned num;
+  unsigned denom;
+};
+
+class HWRocJpegDecoder : public Decoder {
+   public:
+    //! Default constructor
+    HWRocJpegDecoder();
+#if ENABLE_HIP
+    HWRocJpegDecoder(hipStream_t &stream) { _hip_stream = stream; }
```

Comment:
Is there an option to use HWRocJpegDecoder without HIP? If not, the entire class definition should be within ENABLE_HIP if guard

Reply:
Done